src/ray/common/status.cc

```diff
@@ -36,79 +36,47 @@
 
 namespace ray {
 
-#define STATUS_CODE_OK "OK"
-#define STATUS_CODE_OUT_OF_MEMORY "Out of memory"
-#define STATUS_CODE_KEY_ERROR "Key error"
-#define STATUS_CODE_TYPE_ERROR "Type error"
-#define STATUS_CODE_INVALID "Invalid"
-#define STATUS_CODE_IO_ERROR "IOError"
-#define STATUS_CODE_UNKNOWN_ERROR "Unknown error"
-#define STATUS_CODE_NOT_IMPLEMENTED "NotImplemented"
-#define STATUS_CODE_REDIS_ERROR "RedisError"
-#define STATUS_CODE_TIMED_OUT "TimedOut"
-#define STATUS_CODE_INTERRUPTED "Interrupted"
-#define STATUS_CODE_INTENTIONAL_SYSTEM_EXIT "IntentionalSystemExit"
-#define STATUS_CODE_UNEXPECTED_SYSTEM_EXIT "UnexpectedSystemExit"
-#define STATUS_CODE_CREATION_TASK_ERROR "CreationTaskError"
-#define STATUS_CODE_NOT_FOUND "NotFound"
-#define STATUS_CODE_DISCONNECTED "Disconnected"
-#define STATUS_CODE_SCHEDULING_CANCELLED "SchedulingCancelled"
-#define STATUS_CODE_ALREADY_EXISTS "AlreadyExists"
-#define STATUS_CODE_OBJECT_EXISTS "ObjectExists"
-#define STATUS_CODE_OBJECT_NOT_FOUND "ObjectNotFound"
-#define STATUS_CODE_OBJECT_ALREADY_SEALED "ObjectAlreadySealed"
-#define STATUS_CODE_OBJECT_STORE_FULL "ObjectStoreFull"
-#define STATUS_CODE_TRANSIENT_OBJECT_STORE_FULL "TransientObjectStoreFull"
-#define STATUS_CODE_OUT_OF_DISK "OutOfDisk"
-#define STATUS_CODE_OBJECT_UNKNOWN_OWNER "ObjectUnknownOwner"
-#define STATUS_CODE_RPC_ERROR "RpcError"
-#define STATUS_CODE_OUT_OF_RESOURCE "OutOfResource"
-#define STATUS_CODE_OBJECT_REF_END_OF_STREAM "ObjectRefEndOfStream"
-#define STATUS_CODE_AUTH_ERROR "AuthError"
-#define STATUS_CODE_INVALID_ARGUMENT "InvalidArgument"
-#define STATUS_CODE_CHANNEL_ERROR "ChannelError"
-#define STATUS_CODE_CHANNEL_TIMEOUT_ERROR "ChannelTimeoutError"
-
+const std::string STATUS_CODE_OK = "OK";
 // not a real status (catch all for codes not known)
-#define STATUS_CODE_UNKNOWN "Unknown"
+const std::string STATUS_CODE_UNKNOWN = "Unknown";
```

Comment:
Constant variables should be named as `kStatusCodeUnknown`.
Checkout https://google.github.io/styleguide/cppguide.html#Constant_Names

Reply:
And it would be nice if you could use `constexpr std::string_view`, since global trivially destructible objects are strongly discouraged: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
> Constant variables should be named as kStatusCodeUnknown.

ACK

> And it would be nice if you could use constexpr std::string_view, since global trivially destructible objects are strongly discouraged: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

I wrote the reason why I don't use `constexpr std::string_view` in this PR. Does it still make sense to use `string_view` if we always need to convert it into a string?

Reply:
I'm ok with string or string view.

string view is acceptable here since the content is pretty small so it fits in SSO (Short String Optimization) buffer so no allocation.

Reply:
updated